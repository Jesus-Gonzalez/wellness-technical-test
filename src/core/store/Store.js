import React from 'react'
import PropTypes from 'prop-types'
import { applyMiddleware, createStore, compose } from 'redux'
import { Provider } from 'react-redux'
import createSagaMiddleware from 'redux-saga'

import reducers from './reducers'
import sagas from './sagas'

const sagaMiddleware = createSagaMiddleware()

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose

const store = createStore(reducers, composeEnhancers(
  applyMiddleware(sagaMiddleware)
))

sagaMiddleware.run(sagas)

const propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.element
  ]).isRequired
}

export const Store = ({ children }) => {
  return (
    <Provider store={store}>
      {children}
    </Provider>
  )
}

Store.propTypes = propTypes
