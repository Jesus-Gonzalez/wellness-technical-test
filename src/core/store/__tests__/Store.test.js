import React from 'react'
import renderer from 'react-test-renderer'

import { Store } from '../Store'

describe('redux', () => {
  it('Store should be created', () => {
    const Component = renderer.create(<Store>testing</Store>)

    expect(Component).toMatchSnapshot()
  })
})
