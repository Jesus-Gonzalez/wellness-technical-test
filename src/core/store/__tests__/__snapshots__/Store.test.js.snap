// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux Store should be created 1`] = `"testing"`;
